typescript/http-registry-server/HttpServer.ts

```diff
@@ -0,0 +1,129 @@
+import express, { Express, Request, Response } from 'express';
+import type { Logger } from 'pino';
+
+import { IRegistry } from '@hyperlane-xyz/registry';
+import { createServiceLogger } from '@hyperlane-xyz/utils';
+
+import packageJson from './package.json' with { type: 'json' };
+import { AppConstants, ServerConstants } from './src/constants/index.js';
+import { createErrorHandler } from './src/middleware/errorHandler.js';
+import { createChainRouter } from './src/routes/chain.js';
+import { createRootRouter } from './src/routes/root.js';
+import { createWarpRouter } from './src/routes/warp.js';
+import { ChainService } from './src/services/chainService.js';
+import { RegistryService } from './src/services/registryService.js';
+import { RootService } from './src/services/rootService.js';
+import { WarpService } from './src/services/warpService.js';
+
+export class HttpServer {
+  app: Express;
+  protected readonly logger: Logger;
+
+  private constructor(
+    protected getRegistry: () => Promise<IRegistry>,
+    logger: Logger,
+  ) {
+    this.logger = logger;
+    this.app = express();
+    this.app.set('trust proxy', true); // trust proxy for x-forwarded-for header
+    this.app.use(express.json());
+  }
+
+  static async create(
+    getRegistry: () => Promise<IRegistry>,
+  ): Promise<HttpServer> {
+    const logger = await createServiceLogger({
+      service: 'http-registry-server',
+      version: packageJson.version,
+    });
+    return new HttpServer(getRegistry, logger);
+  }
+
+  async start(
+    portInput = process.env.PORT,
+    refreshIntervalInput = process.env.REFRESH_INTERVAL,
+  ) {
+    let port = parseInt(portInput || '', 10);
+    if (isNaN(port)) {
+      if (portInput) {
+        this.logger.warn(
+          { port: portInput, defaultPort: ServerConstants.DEFAULT_PORT },
+          `Invalid PORT value "${portInput}". Falling back to default ${ServerConstants.DEFAULT_PORT}.`,
+        );
+      }
+      port = ServerConstants.DEFAULT_PORT;
```

Comment:
nit: Why not do this instead to avoid nesting ifs?
```typescript
    const parsedPort = parseInt(portInput || '', 10)
    const isNumber = isNaN(port) 
    const port = isNumber ? parsedPort : ServerConstants.DEFAULT_PORT;
    if (!isNumber && portInput) {
        this.logger.warn(
          { port: portInput, defaultPort: ServerConstants.DEFAULT_PORT },
          `Invalid PORT value "${portInput}". Falling back to default ${ServerConstants.DEFAULT_PORT}.`,
        );
      }

```